Quotient/room.cpp

```diff
@@ -123,11 +126,22 @@ class Q_DECL_HIDDEN Room::Private {
     // Starting up with estimate event statistics as there's zero knowledge
     // about the timeline.
     EventStats partiallyReadStats {}, unreadStats {};
+
+    // Legacy for User objects remove when this is deprecated for user in context.
     members_map_t membersMap;
+    QList<User*> membersLeft;
     QList<User*> usersTyping;
-    QHash<QString, QSet<QString>> eventIdReadUsers;
     QList<User*> usersInvited;
-    QList<User*> membersLeft;
+
+    // Instances of RoomMember to replace use of User.
+    QHash<QString, QSharedPointer<RoomMember>> joinedMembers;
+    QHash<QString, QSharedPointer<RoomMember>> invitedMembers;
+    QHash<QString, QSharedPointer<RoomMember>> knockingMembers;
+    QHash<QString, QSharedPointer<RoomMember>> leftMembers;
+    QHash<QString, QSharedPointer<RoomMember>> bannedMembers;
+    QHash<QString, QSharedPointer<RoomMember>> membersTyping;
```

Comment:
This is from an older commit I believe as I actually already removed them. Since it's a value class I didn't see the point in storing them at all as that would just create unneeded copies as they were passed around by value and creating them on command is pretty light anyway.

Reply:
Oh ok, I seem to have got lost in the commits on GitHub's mobile interface...